src/filter_iterator.rs

```diff
@@ -0,0 +1,83 @@
+use crate::iter::KeyValueIterator;
+use crate::types::RowEntry;
+use crate::utils::is_not_expired;
+use crate::SlateDBError;
+
+pub(crate) struct FilterIterator<T: KeyValueIterator> {
+    iterator: T,
+    predicate: Box<dyn Fn(&RowEntry) -> bool + Send>,
+}
+
+impl<T: KeyValueIterator> FilterIterator<T> {
+    pub(crate) fn new(iterator: T, predicate: Box<dyn Fn(&RowEntry) -> bool + Send>) -> Self {
```

Comment:
I added the `+ Send` here to fix a clippy error, IIUC this is essentially marking the RowEntry safe to move across threads, is that correct (and is this actually right/is it safe?)